src/Html.php

```diff
@@ -4788,106 +4788,22 @@ public static function jsAdaptDropdown($id, $params = [])
-               }
-
-               // Skip if there is no 'children' property
-               if (typeof data.children === 'undefined') {
-                  var match  = fuzzy.match(searched_term, data_text, select2_fuzzy_opts);
-                  if (match == null) {
-                     return false;
-                  }
-                  data.rendered_text = match.rendered_text;
-                  data.score = match.score;
-                  return data;
-               }
-
-               // `data.children` contains the actual options that we are matching against
-               // also check in `data.text` (optgroup title)
-               var filteredChildren = [];
-
-               $.each(data.children, function (idx, child) {
-                  var child_text = typeof(child.text) === 'string'
-                     ? getTextWithoutDiacriticalMarks(child.text)
-                     : '';
-
-                  if (child_text.indexOf('>') !== -1 || child_text.indexOf('<') !== -1) {
-                     // escape text, if it contains chevrons (can already be escaped prior to this point :/)
-                     child_text = jQuery.fn.select2.defaults.defaults.escapeMarkup(child_text);
-                  }
-
-                  var match_child = fuzzy.match(searched_term, child_text, select2_fuzzy_opts);
-                  var match_text  = fuzzy.match(searched_term, data_text, select2_fuzzy_opts);
-                  if (match_child !== null || match_text !== null) {
-                     if (match_text !== null) {
-                        data.score         = match_text.score;
-                        data.rendered_text = match_text.rendered;
-                     }
-
-                     if (match_child !== null) {
-                        child.score         = match_child.score;
-                        child.rendered_text = match_child.rendered;
-                     }
-                     filteredChildren.push(child);
-                  }
-               });
-
-               // If we matched any of the group's children, then set the matched children on the group
-               // and return the group object
-               if (filteredChildren.length) {
-                  var modifiedData = $.extend({}, data, true);
-                  modifiedData.children = filteredChildren;
+        $js = <<<JS
+            select2_configs['{$id}'] = {
+                type: 'adapt',
+                field_id: '{$id}',
+                width: '{$width}',
+                placeholder: '{$placeholder}',
+                ajax_limit_count: {$CFG_GLPI['ajax_limit_count']},
+                templateresult: {$templateresult},
+                templateselection: {$templateselection}
+            };
+JS;
 
-                  // You can return modified objects from here
-                  // This includes matching the `children` how you want in nested data sets
-                  return modifiedData;
-               }
+        if ($params['init']) {
```

Comment:
This is a new parameter. Even if there is no usage of the `Html::jsAdaptDropdown()` outside of the `Dropdown::showFromArray()` method, you should ensure that calling this method without defining `$params['init']` will have the same behaviour as the previous code.

```suggestion
        if ($params['init'] ?? true) {
```

Also, you will have to explicitely pass `'init' => false` if you do not want this dropdown to be initialized.